### PR TITLE
Matching Fragments where id and pages are both matched

### PR DESCRIPTION
**Pull request by yCobanoglu**

### CODE REVIEW

**Comment by yCobanoglu** [comment]
[Search Bibliography](https://trello.com/c/22hMomoO/319-search-bibliography)

File: ebl/fragmentarium/infrastructure/fragment_repository.py [hunk 1]
```diff
@@ -54,19 +54,21 @@ def query_by_fragment_number(self, number):
...
+                "id": ref_id,
+            }
         }
-        if reference_pages:
-            match["references.pages"] = {
-                "$regex": fr"^[^\d]*{reference_pages}[^\d]*$"
+        if ref_pages:
+            match["references"]["$elemMatch"]["pages"] = {
+                "$regex":
+                    fr"(^(no.\s*)?{ref_pages}-?[\d]*|^(no.\s*)?[\d]*-?{ref_pages}*)"
```
Comment (jlaasonen) [inline ebl/fragmentarium/infrastructure/fragment_repository.py:69]: Why is `[^\d]*` changed to `(no.\s*)?`?

File: ebl/fragmentarium/application/fragment_repository.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def query_by_fragment_number(self, number: FragmentNumber) -> Fragment:
 
     @abstractmethod
     def query_by_id_and_page_in_references(
-            self, reference_id: str, reference_pages: str) -> List[Fragment]:
+            self, ref_id: str, ref_pages: str) -> List[Fragment]:
```
[the remark this answers is not in the document]
Comment (jlaasonen) [reply ebl/fragmentarium/application/fragment_repository.py:28]: If you are concerned by too-long variable names, you can drop the prefix completely. Reference is already mentioned in the method and having it in all the arguments is redundant.

File: ebl/fragmentarium/application/fragment_repository.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def query_by_fragment_number(self, number: FragmentNumber) -> Fragment:
 
     @abstractmethod
     def query_by_id_and_page_in_references(
-            self, reference_id: str, reference_pages: str) -> List[Fragment]:
+            self, _id: str, pages: str) -> List[Fragment]:
```
Comment (jlaasonen) [inline ebl/fragmentarium/application/fragment_repository.py:28]: Use `id_` for name conflicts and `_id` for unused arguments. Mongo property `_id` should not lead outside of the repositories.

**Pull request merged**